src/java/org/jivesoftware/openfire/user/User.java

```diff
@@ -472,28 +436,34 @@ public String put(String key, String value) {
             String answer;
             String keyString = key;
 
-            synchronized (getName() + keyString.intern()) {
-                if (properties.containsKey(keyString)) {
-                    String originalValue = properties.get(keyString);
-                    answer = properties.put(keyString, value);
-                    updateProperty(keyString, value);
-                    // Configure event.
-                    eventParams.put("type", "propertyModified");
-                    eventParams.put("propertyKey", key);
-                    eventParams.put("originalValue", originalValue);
-                }
-                else {
-                    answer = properties.put(keyString, value);
-                    insertProperty(keyString, value);
-                    // Configure event.
-                    eventParams.put("type", "propertyAdded");
-                    eventParams.put("propertyKey", key);
+            try {
+                synchronized (getName() + keyString.intern()) {

```

Comment:
Is this right? I think this would result in a non-interned String, which would be a temporary object here, and so not syncrhonize properly. Appreciate this is existing code, but might be nice to fix if needed.